File: rb/features.rb
# OfficialJoke SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module OfficialJokeFeatures
  def self.make_feature(name)
    case name
    when "base"
      OfficialJokeBaseFeature.new
    when "ratelimit"
      OfficialJokeRatelimitFeature.new
    when "retry"
      OfficialJokeRetryFeature.new
    when "test"
      OfficialJokeTestFeature.new
    when "timeout"
      OfficialJokeTimeoutFeature.new
    else
      OfficialJokeBaseFeature.new
    end
  end
end

File: py/features.py
# OfficialJoke SDK feature factory

from feature.base_feature import OfficialJokeBaseFeature
from feature.test_feature import OfficialJokeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OfficialJokeBaseFeature(),
        "test": lambda: OfficialJokeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
